configuration_go/schemas/thanos/cache/cache.go

```diff
@@ -0,0 +1,35 @@
+package cache
+
+import (
+	"github.com/bwplotka/mimic"
+	"github.com/observatorium/observatorium/configuration_go/schemas/thanos/cache/memcached"
+	"github.com/observatorium/observatorium/configuration_go/schemas/thanos/cache/memory"
+	"github.com/observatorium/observatorium/configuration_go/schemas/thanos/cache/redis"
+	"gopkg.in/yaml.v2"
+)
+
+// IndexCacheConfig specifies the index cache config.
+type IndexCacheConfig struct {
+	ConfigType string      `yaml:"type"`
+	Config     interface{} `yaml:"config"`
+}
+
+func (c IndexCacheConfig) String() string {
+	ret, err := yaml.Marshal(c)
+	if err != nil {
+		mimic.Panicf("error mashalling IndexCacheConfig to yaml: %v", err)
```

Comment:
Feels a bit weird that a method to transform something to a string needs a very specific helper from a third party library to panic and all the helper does is `panic(fmt.Sprintf("mimic: "+format, a...))`. I think it's something we could think about in the future. It raises an eyebrow but it's fine for now. 🤔

Reply:
Yes I agree, I would rather simply use the standard panic without any ref to mimic. There were some other `mimic.Panic` in other places, followed blindly this convention. Will change.

Reply:
@saswatamcode are you ok with this change?

Reply:
Sure! As long as we panic, I'm good

Reply:
done